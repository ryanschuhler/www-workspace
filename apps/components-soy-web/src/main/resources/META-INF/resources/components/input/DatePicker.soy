{namespace DatePicker}

/**
 * DatePicker component
 */
{template .render}
  {@param id: string}
  {@param? dateFormat: string}
  {@param? maxDate: string}
  {@param? minDate: string}
  {@param? cssClass: string}
  {@param? value: string}
  {@param? focusOnInput: any|null}


  {let $inputAttributes kind="attributes"}
    {if $cssClass}
      class="{$cssClass}"
    {/if}

    {if $dateFormat}
      data-date-format="{$dateFormat}"
    {/if}

    {if $maxDate}
      data-max-date="{$maxDate}"
    {/if}


    {if $minDate}
      data-min-date="{$minDate}"
    {/if}

    id="{$id}"
    name="{$id}"
    ref="input"
    type="text"
    value="{isNonnull($value) ? $value : ''}"
  {/let}
  <div class="date-picker">
    <i class="material-icons date-icon" onClick="{$focusOnInput}">insert_invitation</i>
    <input {$inputAttributes}/>
    <span class="arrow"></span>
  </div>
{/template}


