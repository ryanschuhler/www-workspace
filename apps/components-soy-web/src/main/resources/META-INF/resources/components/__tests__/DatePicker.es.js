'use strict';

import flatpickr from 'flatpickr';

import DatePicker from '../input/DatePicker.es';

let component;

describe('DatePicker', () => {
  afterEach(() => {
    if (component) {
      component.dispose();
    }
  });

  beforeAll(() => {
		window.Liferay = {
      Loader: {
        require: jest.fn()
      }
    };
  });

  it('should render', () => {
    component = new DatePicker({
      id: 'foo'
    });

    component.flatpickr = new flatpickr(component.refs['input']);

    expect(component).toMatchSnapshot();
  });

  it('should render with date format m/d/Y', () => {
    component = new DatePicker({
      dateFormat: 'm/d/Y',
      id: 'foo'
    });

    let input = component.refs['input'];

    component.flatpickr = new flatpickr(input);

    expect(input.getAttribute('data-date-format')).toBe('m/d/Y');
  });
});