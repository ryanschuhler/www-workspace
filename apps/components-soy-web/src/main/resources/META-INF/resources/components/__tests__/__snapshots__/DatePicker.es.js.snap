// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker should render 1`] = `
<div class="date-picker">
  <i class="material-icons date-icon">insert_invitation</i>
  <input id="foo" name="foo" ref="input" type="text" value="" class="flatpickr-input" readonly="readonly">
  <span class="arrow"></span>
</div>
`;
